6.Knative_with_Cloud_Resource_App/README.md

```diff
@@ -0,0 +1,232 @@
+`component-knative-configuration-flask-app.yaml`.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: webapp
+spec:
+  workload:
+    apiVersion: serving.knative.dev/v1
+    kind: Configuration
+    metadata:
+      name: webapp
+      namespace: default
+    spec:
+      template:
+        spec:
+          containers:
+            - image: zzxwill/postgresql-flask-web-application:v0.1
+              name: webapp
+              env:
+                - name: DB_HOST
+                  valueFrom:
+                    secretKeyRef:
+                      name: db-secret
+                      key: endpoint
+                - name: DB_USER
+                  valueFrom:
+                    secretKeyRef:
+                      name: db-secret
+                      key: username
+                - name: DB_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: db-secret
+                      key: password
+              ports:
+                - containerPort: 80
+                  name: http1 # Must be one of "http1" or "h2c" (if supported). Defaults to "http1".
+
+```
+
+## Deploy Application
+Compose cloud resource Component `db` and front web application Component `webapp` as ApplicationConfiguration
+`knative-postgresql-appconfig` in `appconfig-1.yaml`.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: knative-postgresql-appconfig
+spec:
+  components:
+    - componentName: webapp
+
+    - componentName: db
+```
+
+In a Kubernetes cluster, submit manifests of these two components and the ApplicationConfiguration.
+
+```shell
+✗ kubectl apply -f component-postgresql.yaml
+component.core.oam.dev/db created
+
+✗ kubectl apply -f component-knative-configuration-flask-app.yaml
+component.core.oam.dev/webapp created
+
+✗ kubectl apply -f appconfig-1.yaml
+applicationconfiguration.core.oam.dev/knative-postgresql-appconfig configured
+```
+
+By checking the list of Component, two components `db` and `webapp` are provisioned. After several minutes,
+the statue of RDSInstance will become to `Running`.
+```shell
+✗ k get component
+NAME        WORKLOAD-KIND
+db          RDSInstance
+webapp      Configuration
+
+✗ k get RDSInstance
+NAME             READY   SYNCED   STATE      ENGINE       VERSION   AGE
+rdspostgresql3   True    True     Running    PostgreSQL   9.4       11m
+```
+At this time, a Postgresql instance is provisioned, and we can see it from Alibaba Clod console.
+![image.png](./postgresql_instance.png)<br />
+
+The application `webapp` also generates a Knative Configuration.
+
+```shell
+✗ k get configuration
+NAME     LATESTCREATED   LATESTREADY    READY   REASON
+webapp   webapp-c8s4n    webapp-c8s4n   True
+```
+
+Edit `appconfig-1.yaml`, add Knative Route with the latest revision of Configuration as shown in `appconfig-2.yaml` and submit it.
```

Comment:
This step makes the demo break. We need dependency feature \cc @hongchaodeng

Reply:
Will use datainput/dataoutput mechanism after the [PR](https://github.com/crossplane/oam-kubernetes-runtime/pull/48) is merged.

Reply:
Please hold on the code merging, I hit an issue for verifying this App with the pending PR https://github.com/crossplane/oam-kubernetes-runtime/pull/48#issuecomment-648655426.